src/strategies/strategies/arrow-vesting/index.ts

```diff
@@ -38,59 +38,154 @@ export async function strategy(
+      let batchResults;
+      let retries = 3;
+      while (retries > 0) {
+        try {
+          const vestingFactoryMulti = new Multicaller(
+            network,
+            provider,
+            vestingFactoryAbi,
+            { blockTag }
+          );
+
+          for (
+            let contractIdx = startIdx;
+            contractIdx < endIdx;
+            contractIdx++
+          ) {
+            vestingFactoryMulti.call(
+              contractIdx,
+              options.vestingFactory,
+              'escrows',
+              [contractIdx]
+            );
+          }
+
+          batchResults = await vestingFactoryMulti.execute();
+          break; // Success, exit retry loop
+        } catch (error) {
+          retries--;
+          if (retries === 0) {
+            console.warn(
+              `Failed to fetch vesting contracts batch ${startIdx}-${endIdx} after 3 retries:`,
+              error instanceof Error ? error.message : String(error)
+            );
+            // Continue with next batch instead of throwing
+            batchResults = {};
+          } else {
+            // Wait before retry
+            await new Promise(resolve => setTimeout(resolve, 1000));
+          }
+        }
+      }
+
+      return batchResults;
+    })();
+
+    batchPromises.push(batchPromise);
+  }
+
+  // Wait for all batches to complete and merge results
+  const batchResults = await Promise.all(batchPromises);
+  batchResults.forEach(results => Object.assign(vestingContracts, results));
+
+  // Get all vesting contract parameters in parallel batches
+  const vestingContractAddresses = Object.values(vestingContracts);
+  const vestingContractParameters: Record<string, object> = {};
+
+  const parameterBatchPromises: Promise<Record<string, object>>[] = [];
+  for (
+    let startIdx = 0;
+    startIdx < vestingContractAddresses.length;
+    startIdx += batchSize
+  ) {
+    const endIdx = Math.min(
+      startIdx + batchSize,
+      vestingContractAddresses.length
     );
-  });
-
-  const vestingContractParameters: Record<string, object> =
-    await vestingContractMulti.execute();
+    const batchAddresses = vestingContractAddresses.slice(startIdx, endIdx);
+
+    const parameterBatchPromise = (async () => {
+      let batchResults;
+      let retries = 3;
+
+      while (retries > 0) {
+        try {
+          const vestingContractMulti = new Multicaller(
+            network,
+            provider,
+            vestingContractAbi,
+            { blockTag }
+          );
+
+          batchAddresses.forEach(vestingContractAddress => {
+            vestingContractMulti.call(
+              `${vestingContractAddress}.recipient`,
+              vestingContractAddress,
+              'recipient',
+              []
+            );
+            vestingContractMulti.call(
+              `${vestingContractAddress}.total_locked`,
+              vestingContractAddress,
+              'total_locked',
+              []
```

Comment:
I don't think this gets used later in the function so we could remove it from the multicall.